pkg/shared/families/malware/yara/yara.go

```diff
@@ -0,0 +1,168 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yara
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+
+	"github.com/openclarity/kubeclarity/shared/pkg/job_manager"
+	"github.com/openclarity/kubeclarity/shared/pkg/utils"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/openclarity/vmclarity/pkg/shared/families/malware/common"
+	"github.com/openclarity/vmclarity/pkg/shared/families/malware/yara/config"
+	"github.com/openclarity/vmclarity/pkg/shared/families/malware/yara/util"
+	sharedutils "github.com/openclarity/vmclarity/pkg/shared/utils"
+)
+
+const (
+	ScannerName               = "yara"
+	TemporaryYaraRuleFilePath = "/tmp/yara-rules.yarac"
```

Comment:
Shouldn't it be configurable?

Reply:
I think we should use the golang Mktmp etc functions for temporary directories